config/crd/bases/ocs.openshift.io_autostoragescalings.yaml

```diff
@@ -0,0 +1,127 @@
+                description: DeviceClassName is the name of the device class for which
+                  the storage scaling is to be done.
+                type: string
+              storageClusterName:
+                description: StorageClusterName is the name of the storage cluster
+                  for which the storage scaling is to be done.
+                type: string
+            required:
+            - capacityLimit
+            - deviceClassName
+            - storageClusterName
+            type: object
+          status:
+            description: AutoStorageScalingStatus defines the observed state of AutoStorageScaling
+            properties:
+              expectedOsdCount:
+                description: |-
+                  The Expected OSD count is the count that the auto-expansion has decided to set.
+                  This will be set on the storageCLuster CR as the desired count of the OSDs.
+                  Used for horizontal scaling of OSDs.
+                format: int32
+                type: integer
+              expectedOsdSize:
+                anyOf:
+                - type: integer
+                - type: string
+                description: |-
+                  The ExpectedOsdSize is the size that the auto-expansion has decided to set.
+                  This will be set on the storageCLuster CR as the desired size of the OSDs.
+                  Used for vertical scaling of OSDs.
+                pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
+                x-kubernetes-int-or-string: true
+              knownOsdCount:
+                description: |-
+                  The known OSD count is the original count of the OSDs before the expansion in progress is completed.
+                  After the expansion is completed, this would be updated to the expected OSD count.
+                  Used for horizontal scaling of OSDs.
+                format: int32
+                type: integer
+              knownOsdSize:
+                anyOf:
+                - type: integer
+                - type: string
+                description: |-
+                  The known size is the original size of the OSDs before the expansion in progress is completed.
+                  After the expansion is completed, this would be updated to the expected OSD size.
+                  Used for vertical scaling of OSDs.
+                pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
+                x-kubernetes-int-or-string: true
+              lastRunTimeStamp:
+                description: LastRunTimeStamp is the time stamp of the last run of
+                  the storage scaling
+                format: date-time
+                type: string
+              phase:
+                description: Phase describes the Phase of AutoStorageScaling
```

Comment:
@parth-gr Shouldn't we describe the `enum` allowed values: `waiting,inProgress,completed,failed` ?

Reply:
Its the controller that would be updating the status values, so I don't think we need enum here, wdyt?

Reply:
Right, it's not strictly necessary. My suggestion intended to have a more descriptive explanation for users just like the `StorageCluster` CRD does for `deviceType`:
```
        deviceType:
          description: |-
            DeviceType is the value of device type in
            this StorageDeviceSet. It can have one of the
            three values (SSD, HDD, NVMe)
          enum:
          - SSD
          - ssd
          - HDD
          - hdd
          - NVMe
          - NVME
          - nvme
          type: string
```

Reply:
done, with constant